lib/config/wrapper.go

```diff
@@ -445,3 +445,39 @@ func (w *Wrapper) MyName() string {
 func (w *Wrapper) CheckHomeFreeSpace() error {
 	return checkFreeSpace(w.Options().MinHomeDiskFree, fs.NewFilesystem(fs.FilesystemTypeBasic, filepath.Dir(w.ConfigPath())))
 }
+
+// GenerateClusterConfig returns a protocol.ClusterConfig for the given peer device
+// with all the relevant configuration information. It still needs to be completed
+// with database information.
+func (w *Wrapper) GenerateClusterConfig(device protocol.DeviceID) protocol.ClusterConfig {
```

Comment:
This should not live in the config.

Reply:
As all it does is translate the config accessing unimported infos I considered it a good fit. The other option would be to add something like `FolderSlice() []FolderConfiguration { return w.cfg.Folders }` to `Wrapper`.

Reply:
Config is about config, ClusterConfig is a protocol level thing. If you want, have a helper in the model or something.